mmaction/models/heads/bbox_head.py

```diff
@@ -177,22 +199,29 @@ def loss(self,
              reduce=True):
 
         losses = dict()
+        # Only use the cls_score
         if cls_score is not None:
-            # Only use the cls_score
-            labels = labels[:, 1:]
+            labels = labels[:, 1:]  # Get valid labels (ignore first)
             pos_inds = torch.sum(labels, dim=-1) > 0
             cls_score = cls_score[pos_inds, 1:]
             labels = labels[pos_inds]
 
-            bce_loss = F.binary_cross_entropy_with_logits
+            # Branch on multi/single label
+            #   Note that both losses compute the sigmoid/softmax before
+            if self.multilabel:
+                loss_func = F.binary_cross_entropy_with_logits
+            else:
+                loss_func = F.cross_entropy
 
-            loss = bce_loss(cls_score, labels, reduction='none')
+            # Compute loss
+            loss = loss_func(cls_score, labels, reduction='none')
             pt = torch.exp(-loss)
             F_loss = self.focal_alpha * (1 - pt)**self.focal_gamma * loss
             losses['loss_action_cls'] = torch.mean(F_loss)
 
-            recall_thr, prec_thr, recall_k, prec_k = self.multi_label_accuracy(
-                cls_score, labels, thr=0.5)
+            recall_thr, prec_thr, recall_k, prec_k = self.topk_accuracy(
```

Comment:
BTW, if we use the single label, what about directly reporting the normal Top-k accuracy instead (Top-k: if top k predictions match the ground-truth label)? In that situation, I think we don't need to report accuracy-recall.

Reply:
I guess you are referring to computing just accuracy rather than prec/recall.

It does make sense, although:
 1. I wanted to retain continuity with the previous methods as far as possible (i.e. minimal changes) - doing so would require more changes throughout the library to handle a different metric to be reported
 2. There is still merit to having precision/recall metrics even in in single-label classification (see e.g. https://www.analyticsvidhya.com/blog/2020/12/accuracy-and-its-shortcomings-precision-recall-to-the-rescue/)